givenergy_modbus/client/client.py

```diff
@@ -153,6 +153,97 @@ async def watch_plant(
                     reqs, timeout=timeout, retries=retries, return_exceptions=True
                 )
 
+    async def detect_plant(
```

Comment:
I'm currently aiming towards keeping all this knowledge in plant.py rather than the client having to worry about it.
https://github.com/divenal/givenergy-modbus-async/commit/ea41f0c92d6ce16ea7b9bf3bd4656be28b318984  has code to implement refresh_plant() that way. I hope for detect_plant() to do something similar, where it asks the plant to suggest registers that need to be fetched. Probably iteratively, since we need to identify the model first. (I'm assuming we first determine model by getting HR(0..59) from address 0x11)

That change also adds a mechanism which might be useful for GivTCP Application code can extend Plant class and hook method registers_updated(). Then each time a WriteHoldingRegisterResponse arrives (due to a write from any connected client, or the portal), that method will be invoked. This way, GivTCP doesn't have to poll holding registers regularly to discover what has changed - it can be told immediately.